pkg/trace/config/apply.go

```diff
@@ -358,6 +368,62 @@ func (c *AgentConfig) applyDatadogConfig() error {
 		MaxRequestBytes: c.MaxRequestBytes,
 	}
 
+	// Instrumentation Telemetry config
+	c.TelemetryConfig = &TelemetryConfig{
+		Enabled: config.Datadog.GetBool("apm_config.telemetry.enabled"),
+	}
+	if c.TelemetryConfig.Enabled {
+		site := config.Datadog.GetString("site")
+		if site == "" {
+			site = defaultSite
+		}
+
+		main := telemetryEndpointPrefix + site
+		if v := config.Datadog.GetString("apm_config.telemetry.dd_url"); v != "" {
+			main = v
+		}
+
+		u, err := url.Parse(main)
+		if err != nil {
+			// if the main intake URL is invalid we don't use additional endpoints
+			log.Errorf("error parsing main telemetry intake URL %s: %v", main, err)
+			c.TelemetryConfig.Enabled = false
+		} else {
+			c.TelemetryConfig.Endpoints = []*Endpoint{{
+				APIKey: c.Endpoints[0].APIKey,
+				Host:   u.Host,
+			}}
+		}
+
+		additionalEndpoitnsCfg := "apm_config.telemetry.additional_endpoints"
+
+		if config.Datadog.IsSet(additionalEndpoitnsCfg) {
+			extra := config.Datadog.GetStringMapStringSlice(additionalEndpoitnsCfg)
+			for endpoint, keys := range extra {
+				u, err := url.Parse(endpoint)
+
```

Comment:
If we're going to use `additional_endpoints`, the logic should be the same as elsewhere. Can we simply duplicate it or create a function which does it? Maintaining backwards compatibility. Check https://github.com/DataDog/datadog-agent/blob/2373feabd39cf4148fa0fcc7fb2235bebb2dc220/pkg/trace/config/apply.go#L229-L240

Reply:
Ok, I changed plans. Seeing how Endpoint.Host is actually used more as an URL in trace/writer 

I'm now tempted to refactor Endpoint.Host into Endpoint.URL. but thats a story for a separate PR - I think :)